### PR TITLE
Feat/change email

### PR DESCRIPTION
**Pull request by codingfabi**
Closes frontend part of #106

### CODE REVIEW

**Comment by sami1riaz** [comment]
I signed out of an account and was able to go back to this page: 
image

File: frontend/src/components/EmailSwapDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  }
+`
+
+const newEmailFormInitialValues = {
+    newEmail: '',
+    password: ''
+}
+
+const newEmailFormValidationSchema = yup.object({
+    newEmail: yup
```
Comment (sami1riaz) [inline frontend/src/components/EmailSwapDialog.tsx:51]: the error validation is not being used in the form

**Review by sami1riaz — CHANGES_REQUESTED** [review]
Left a few minor comments. We could deal with the "navigating to user profile without being signed in" error separately.

Haven't been able to actually test the email change functionality and error scenarios because there's something wrong with my setup (can't make GQL requests). But if you have already tested the scenarios, then you can merge this once you've dealt with the comments I have left. (I won't be able to look at this again till the end of next week)

File: frontend/pages/user-profile.tsx [hunk 2]
```diff
@@ -38,7 +39,6 @@ const useStyles = makeStyles((theme) => ({
 const ME = gql`
```
[the remark this answers is not in the document]
Comment (codingfabi) [reply frontend/pages/user-profile.tsx]: done

File: frontend/src/components/EmailSwapDialog.tsx (hunk 1 above)
Comment (codingfabi) [reply frontend/src/components/EmailSwapDialog.tsx:51]: I am not really sure what you mean as I think that it is being used. I am merging anyway, lets discuss this another time maybe.

**Comment by codingfabi** [comment]
> I signed out of an account and was able to go back to this page: image

I implemented a `NoAuthorizationComponent` which we can use in these cases

**Pull request merged**